### PR TITLE
Loss function convergence

### PR DESCRIPTION
**Pull request by jeffjennings**
A simple function to test whether the loss function (or any function generically) has converged as it iterates.

### CODE REVIEW

**Review by iancze — COMMENTED** [review]
This looks like a good start! Can you provide an example of how this might be used in practice? 

E.g., how would the [loop](https://mpol-dev.github.io/MPoL/ci-tutorials/optimization.html#iterating-the-training-loop) change to take advantage of this tracker?

This could be either as a docstring to this method or in a test.

**Comment by jeffjennings** [comment]
I was going to use this function in multiple places, but I've been reorganizing local changes and am down to just one call, so I'm going to rename this branch (closing this PR), rename and re-spec the branch. I'm moving this function into a training class.
